src/OpenLoco/src/Windows/ScenarioSelect.cpp

```diff
@@ -481,8 +497,12 @@ namespace OpenLoco::Ui::Windows::ScenarioSelect
         if (scenarioEntry == nullptr)
             return;
 
-        self.info = reinterpret_cast<uintptr_t>(scenarioEntry);
-        self.invalidate();
+        if (self.info != reinterpret_cast<uintptr_t>(scenarioEntry))
+        {
+            self.info = reinterpret_cast<uintptr_t>(scenarioEntry);
+            self.invalidate();
+            _warnOnce = false;
+        }
```

Comment:
This could have caused a slowdown, right? Perhaps this could be the cause of #1021? I've had trouble reproducing it, but this diff reminded me of that issue.

Reply:
I'm not sure the load screen uses this. Could be wrong but think it's only scenario select

Reply:
Ahh no you're right, I'm mixing things up.